app/fee_distribution.go

```diff
@@ -48,6 +49,66 @@ func (vac *ValAddrCache) GetAccAddr(ctx sdk.Context, consAddr sdk.ConsAddress) s
 	return accAddr
 }
 
+func (vac *ValAddrCache) SetDistributionAddr(consAddr sdk.ConsAddress, accAddr sdk.AccAddress) {
+	vac.distributionAddrCache[string(consAddr)] = accAddr
+}
+
+func (vac *ValAddrCache) GetDistributionAddr(ctx sdk.Context, consAddr sdk.ConsAddress) sdk.AccAddress {
+	if value, ok := vac.distributionAddrCache[string(consAddr)]; ok {
+		return value
+	}
+	validator, found := vac.stakeKeeper.GetValidatorByConsAddr(ctx, consAddr)
+	if !found {
+		panic(fmt.Errorf("can't load validator with consensus address %s", consAddr.String()))
+	}
+	distributionAddr := validator.DistributionAddr
+	vac.SetDistributionAddr(consAddr, distributionAddr)
+	return distributionAddr
+}
+
+func distributeFeeBEPHHH(ctx sdk.Context, am auth.AccountKeeper, valAddrCache *ValAddrCache, publishBlockFee bool, stakeKeeper stake.Keeper) (blockFee pub.BlockFee) {
+	// TODO: needs to add BSC ratio Fee
```

Comment:
Do we need this `TODO`?

Reply:
No, we don't.
It's already implemented without touching fee pool part.